troubleshooting.md

```diff
@@ -0,0 +1,131 @@
+# Troubleshooting
+
+Sorry if things are going wrong!
+
+## Update clangd
+
+Every release fixes many bugs, works in more scenarios, and provides clearer
+logs for troubleshooting.
+
+Try to [install the latest major-version release][installation] or even a
+pre-release snapshot before you try anything else!
+
+## Gathering logs
+
+Your editor should make clangd's `stderr` log available (see [installation][]
+for how). These describe the messages exchanged between the editor and `clangd`,
+and what clangd was thinking.
+
+Even if you don't understand the contents, it's tremendously useful to include
+this when filing a bug. If you add `-log=verbose` to the clangd arguments, this
+will include more detail (e.g. the full content of LSP messages).
+
+## Lots of error diagnostics
+
+If your code builds OK but clangd shows lots of error diagnostics, it
+probably can't parse your code very well. Other features will work poorly too:
+
+![Many errors](screenshots/many_errors.png)
+
+### Can't find includes within your project
+
+This usually means that we don't have a compile command for this file.
+`compile_commands.json` should exist in some parent directory and should have
+a valid command. If this file has changed, you'll need to restart clangd to see
+the changes.
+
+The logs can tell you whether `compile_commands.json` was found and what compile
+command was ultimately used.
+
+### Can't find standard library headers (`<map>`, `<stdio.h>` etc)
+
+First, you'll need the standard library (and any other libraries you depend on)
+installed on your system! Even if you don't actually build on this machine,
+clangd needs to parse the headers.
+
+The standard library headers are often found relative to the compiler.
+Try to configure your project with an absolute path to the compiler
+(`/usr/bin/gcc`, not `gcc`).
+The logs (with `-log=verbose`) can tell you the `cc1` command, including which
+paths were searched for the standard library. You can compare this to the output
+of `clang -### <args>`.
+
+If you're using an unusual compiler (e.g. a cross-compiler for a different
+platform) you may want to pass `--query-driver=/path/to/mygcc` to allow clangd
+to extract the include paths from it directly.
+
+### Can't find compiler built-in headers (`<stddef.h>` etc)
+
+These headers are formally part of the standard library, but so closely coupled
+to the parser that they are installed with clangd.
+
+They must be found in `../lib/clang/<version>/include`. If you've built clangd
+from source or installed it from a package, these should be in the right place.
+If you've moved the binary around, move it back!
+
+## Dealing with crashes
+
+If clangd is exiting unexpectedly, this is probably a crashing bug in clangd.
+Understanding it better will help you work around it, and if you can file an
+informative [bug report][] we'll try to fix it for the next release.
+
+The most helpful crash reports have a log including a (symbolic) stack trace,
```

Comment:
the words explaining an informative bug report inspires that we should use a [template issue](https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository).

Reply:
Good idea, I'll set one up. Nothing to do here though, I think.